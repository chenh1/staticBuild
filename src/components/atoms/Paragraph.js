import React from 'react';

const Paragraph = ({styleClass, text}) => {
    return (
        <p className={styleClass}>
            {text}
        </p>
    );
};

export default Paragraph;