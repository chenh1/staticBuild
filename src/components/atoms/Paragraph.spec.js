import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import Paragraph from './Paragraph';

test('renders paragraph tag with text', () => {
    const tree = renderer.create(<Paragraph
        text={["Mock text"]}
        customClass="mock-class" />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});

test('renders paragraph tag with default paragraph-one class', () => {
    const tree = renderer.create(<Paragraph
        text={["Mock text"]} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});

test('renders paragraph tag with embedded link if exists', () => {
    const tree = renderer.create(<Paragraph
        text={[
            "Mock text", 
            {embeddedLink: '/mockUrl', text:'text with embedded link'}, 
            "more mock text"
        ]}
        customClass="mock-class" />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});
