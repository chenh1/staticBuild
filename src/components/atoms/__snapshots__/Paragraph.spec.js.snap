// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders paragraph tag with default paragraph-one class 1`] = `
<p
  className="paragraph-one"
>
  Mock text
</p>
`;

exports[`renders paragraph tag with embedded link if exists 1`] = `
<p
  className="mock-class"
>
  Mock text
  <a
    className="article-link page-link"
    data-link-type="externalLink"
    href="/mockUrl"
  >
    text with embedded link
  </a>
  more mock text
</p>
`;

exports[`renders paragraph tag with text 1`] = `
<p
  className="mock-class"
>
  Mock text
</p>
`;
